cloudify_agent/worker.py

```diff
@@ -260,7 +264,11 @@ def _update_operation_state(self, ctx, common_version):
         if store:
             ctx.update_operation(constants.TASK_STARTED)
         try:
+            ctx.download_deployment_workdir(ctx.deployment.id,
+                                            ctx.deployment.tenant_name)
             yield
+            ctx.upload_deployment_workdir(ctx.deployment.id,
+                                          ctx.deployment.tenant_name)
```

Comment:
possibly both the `upload` clauses should be under a `finally` clause, so that even if an operation was errorneous (e.g. kill-cancelled), the files changed by it, are still synced up